rpc/src/v1/types/uint.rs

```diff
@@ -1,172 +0,0 @@
-
-// You should have received a copy of the GNU General Public License
-// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
-
-use std::str::FromStr;
-use std::fmt;
-use serde;
-use ethereum_types::{U256 as EthU256, U128 as EthU128};
-
-macro_rules! impl_uint {
-	($name: ident, $other: ident, $size: expr) => {
-		/// Uint serialization.
-		#[derive(Debug, Default, Clone, Copy, PartialEq, Hash)]
-		pub struct $name($other);
-
-		impl Eq for $name { }
-
-		impl<T> From<T> for $name where $other: From<T> {
-			fn from(o: T) -> Self {
-				$name($other::from(o))
-			}
-		}
-
-		impl FromStr for $name {
-			type Err = <$other as FromStr>::Err;
-
-			fn from_str(s: &str) -> Result<Self, Self::Err> {
-				$other::from_str(s).map($name)
-			}
-		}
-
-		impl Into<$other> for $name {
-			fn into(self) -> $other {
-				self.0
-			}
-		}
-
-		impl fmt::Display for $name {
-			fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-				write!(f, "{}", self.0)
-			}
-		}
-
-		impl fmt::LowerHex for $name {
-			fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-				fmt::LowerHex::fmt(&self.0, f)
-			}
-		}
-
-		impl<'a> serde::Deserialize<'a> for $name {
-			fn deserialize<D>(deserializer: D) -> Result<$name, D::Error>
-			where D: serde::Deserializer<'a> {
-				struct UintVisitor;
-
-				impl<'b> serde::de::Visitor<'b> for UintVisitor {
-					type Value = $name;
-
-					fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-						write!(formatter, "a 0x-prefixed, hex-encoded number of length {}", $size*16)
-					}
-
-					fn visit_str<E>(self, value: &str) -> Result<Self::Value, E> where E: serde::de::Error {
-						if value.len() < 2  || !value.starts_with("0x") {
-							return Err(E::custom("expected a hex-encoded numbers with 0x prefix"))
-						}
-
-						// 0x + len
-						if value.len() > 2 + $size * 16 {
-							return Err(E::invalid_length(value.len() - 2, &self));
-						}
-
-						$other::from_str(&value[2..]).map($name).map_err(|e| E::custom(&format!("invalid hex value: {:?}", e)))
-					}
-
-					fn visit_string<E>(self, value: String) -> Result<Self::Value, E> where E: serde::de::Error {
-						self.visit_str(&value)
-					}
-				}
-
-				deserializer.deserialize_any(UintVisitor)
-			}
-		}
-
-	}
-}
-
-impl_uint!(U128, EthU128, 2);
-impl_uint!(U256, EthU256, 4);
-impl_uint!(U64, u64, 1);
-
-impl serde::Serialize for U128 {
-	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where S: serde::Serializer {
-		serializer.serialize_str(&format!("{:#x}", self))
-	}
-}
-
-impl serde::Serialize for U256 {
-	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where S: serde::Serializer {
-		serializer.serialize_str(&format!("{:#x}", self))
-	}
-}
-
-impl serde::Serialize for U64 {
-	fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error> where S: serde::Serializer {
-		serializer.serialize_str(&format!("{:#x}", self))
-	}
-}
-
-#[cfg(test)]
-mod tests {
```

Comment:
Can we leave the tests to make sure that our assumptions are not broken when the library is updated? Also does 0.4 contain the fix for non-ascii characters?

Reply:
Moved them to a [new_file](https://github.com/paritytech/parity-ethereum/blob/f319a687aa81d8a7c5f7224726d70778756a5cf0/rpc/src/v1/types/eth_types.rs) ok?

Reply:
yup :ok_hand: